specs/Standards/MultiToken/README.md

```diff
@@ -0,0 +1,539 @@
+
+Let's assume the following:
+- The `gold` token is represented by the `games` with token_id `g133`.
+- Alice's account is `alice`.
+- Bob's account is `bob`.
+- The precision for `gold` on the games contract is `10^8`. 
+- The 5 tokens is `5 * 10^8` or as a number is `500000000`. 
+
+High-level explanation:
+
+Alice needs to issue one transaction to `games` contract to transfer 5 tokens (multiplied by precision) to Bob.
+ 
+Technical calls:
+
+1. `alice` calls `games::mt_transfer({"receiver_id": "bob", "amount": "500000000", "token_id": "g133", "memo": "for my dinner"})`.
+
+#### Simple batch transfer
+
+Alice wants to send and a 1 unique(nft) gemstone, 5 gold and 10 silver tokens to Bob.
+
+Let's assume the following:
+- The unique nft `gem` token is represented by `games` with token_id
+`uu2`
+- The `gold` token is represented by the `games` with token_id `g133`.
+- The `silver` token is represented by the `games` with token_id `s133`.
+- Alice's account is `alice`.
+- Bob's account is `bob`.
+- The precision for `gold` on the games contract is `10^8`. 
+- The precision for `silver` on the games contract is also `10^8`. 
+- The 5 gold tokens is `5 * 10^8` or as a number is `500000000`. 
+- The 10 silver tokens is `10 * 10^8` or as a number is `1000000000`. 
+- The  1 gem token is `1` or as a number is `1`
+
+High-level explanation:
+
+Alice needs to issue one transaction to `games` contract to transfer 5 gold tokens and 10 silver tokens (multiplied by precision) and 1 gem to Bob.
+ 
+Technical calls:
+
+1. `alice` calls `games::mt_transfer_batch({"receiver_id": "bob", "amounts": ["500000000", "1000000000", "1"], "token_ids": ["g133", "s133", "uu2"], "memo": "well done"})`.
+
+
+
+#### Token deposit to a contract 
+
+Alice wants to deposit `gold` tokens to a compound interest contract to earn some rewards.
+
+Let's assume the following:
+- The `gold` token is represented  by the `games` contract with token_id `g133` .
+- Alice's account is `alice`.
+- The compound interest contract is `compound`.
+- The precision on `gold` token is `10^18`. 
+- The 1000 tokens is `1000 * 10^18` or as a number is `1000000000000000000000`. 
+- The compound contract can work with many different token contracts and types.
+
+High-level explanation:
+
+Alice needs to issue a single transaction to `games` that will internally issue a cross contract call to `compound`. 
+
+The initial transaction to `games` is made with `compound` as the receiver of a set token_ids and amounts from `alice`.
+
+This call then waits on a response from `compound`. If `compound` responds with failure, the tx is aborted. 
+
+Otherwise `games` contract accepts the results and resolves the promise completing the transaction.
+
+- If transfer succeeded, `compound` can increase local ownership for `alice` to 1000 for `gold` token_id `g133`
+
+- If transfer fails, `compound` doesn't need to do anything in current example, but maybe can notify `alice` of unsuccessful transfer.
+
```

Comment:
Compound.near is expected to implement the TokenReceiver trait and games.near is expected to make cross contract call to `compound::mt_on_transfer`. games.near will then handle the promise that compound::mt_on_transfer returns. This promise can be a failure response, if compound::mt_on_transfer returns fails. This might look like the following below:  
```
pub fn games.near::made_up_send_tokens() {
compound.near::mt_on_transfer()
.then(ext_self::mt_resolve_transfer(...)) 
}  
```

Reply:
I see a lot of `must` wording around `mt_on_transfer` and `mt_resolve_transfer`, but this example makes it sound like we're talking about a completely optional thing.  Are these two different cases?

Reply:
mt_resolve_transfer can be called anything and isn't strictly required, however the developer is encouraged to evaluate the mt_on_transfer result.   So mt_resolve_transfer is a this is how you handle the result of calling mt_on_transfer. 

Contracts that want to receive tokens are required to implement mt_on_transfer.